demo/llm_hf_transformers/run.py

```diff
@@ -0,0 +1,47 @@
+import gradio as gr
+import torch
+from transformers import AutoModelForCausalLM, AutoTokenizer, StoppingCriteria, StoppingCriteriaList, TextIteratorStreamer
+from threading import Thread
+
+tokenizer = AutoTokenizer.from_pretrained("togethercomputer/RedPajama-INCITE-Chat-3B-v1")
+model = AutoModelForCausalLM.from_pretrained("togethercomputer/RedPajama-INCITE-Chat-3B-v1", torch_dtype=torch.float16)
+model = model.to('cuda:0')
+
+class StopOnTokens(StoppingCriteria):
+    def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor, **kwargs) -> bool:
+        stop_ids = [29, 0]
+        return any(input_ids[0][-1] == stop_id for stop_id in stop_ids)
+
+def predict(message, history):
+    history_transformer_format = list(zip(history[:-1], history[1:])) + [[message, ""]]
+    stop = StopOnTokens()
+
+    messages = "".join(["".join(["\n<human>:"+item[0], "\n<bot>:"+item[1]])
```

Comment:
This code is now outdated. you don't have to apply the chat template yourself since transformers supports the `role/content` dictionary conversation format. 

I think the streaming can also be simplified but not sure.